src/runtime/type.ts

```diff
@@ -1381,6 +1378,17 @@ export class TypeVariableInfo {
     if (this._resolution) {
       return this._resolution.maybeEnsureResolved();
     }
+    if (this._resolveToMaxType) {
+      if (this._canReadSubset) {
+        this.resolution = this._canReadSubset;
+        return true;
+      }
+      if (this._canWriteSuperset) {
```

Comment:
you don't need this if statement or the subsequent `return false`, just fall into the !_resolveToMaxType case.

Reply:
E.g. this will work:
```
  maybeEnsureResolved() {
    if (this._resolution) {
      return this._resolution.maybeEnsureResolved();
    }
    if (this._resolveToMaxType && this._canReadSubset) {
      this.resolution = this._canReadSubset;
      return true;
    }
    if (this._canWriteSuperset) {
      this.resolution = this._canWriteSuperset;
      return true;
    }
    if (this._canReadSubset) {
      this.resolution = this._canReadSubset;
      return true;
    }
    return false;
  }
```

Reply:
That's a good optimization, implemented.